internal/rbd/controllerserver.go

```diff
@@ -896,6 +920,7 @@ func (cs *ControllerServer) ValidateVolumeCapabilities(
 }
 
 // CreateSnapshot creates the snapshot in backend and stores metadata in store.
+// nolint:gocyclo,cyclop // reduce complexity
```

Comment:
add nolint is not encouraged as it gets staled once this PR is merged. common practice is to do refactor and add the functionality.